plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1701,8 +1706,110 @@ public void testTruncateDecimalTransform()
+                "'Grozny'",
+                "'mommy'",
+                "'Greece'",
+                "'moscow'");
+
+        testBucketTransformForType(
+                "BIGINT",
+                ImmutableList.of("100000000", "200000002", "300000004", "700000007"),
+                ImmutableList.of(1, 4, 6, 7),
+                ImmutableList.of("400000001", "500000003", "600000005"),
+                ImmutableList.of(2, 3, 5),
+                "'100000000'",
+                "'700000007'",
+                "'400000001'",
+                "'600000005'");
+    }
+
+    private void testBucketTransformForType(
+            String type,
+            List<String> firstBucketKeys,
+            List<Integer> firstBucketValues,
+            List<String> secondBucketKeys,
+            List<Integer> secondBucketValues,
+            String firstBucketMinKey,
+            String firstBucketMaxKey,
+            String secondBucketMinKey,
+            String secondBucketMaxKey)
+    {
+        String tableName = format("test_bucket_transform%s", type);
+        String select = format("SELECT d_bucket, row_count, d.min AS d_min, d.max AS d_max, b.min AS b_min, b.max AS b_max FROM \"%s$partitions\"", tableName);
+
+        assertUpdate(format("CREATE TABLE %s (d %s, b BIGINT) WITH (partitioning = ARRAY['bucket(d, 2)'])", tableName, type));
+
+        Stream<String> firstBucket = zip(firstBucketKeys.stream(), firstBucketValues.stream(), (key, value) -> format("(%s,%d)", key, value));
+        Stream<String> secondBucket = zip(secondBucketKeys.stream(), secondBucketValues.stream(), (key, value) -> format("(%s,%d)", key, value));
+        String allValues = StringUtils.join(Stream.concat(firstBucket, secondBucket).collect(toList()), ",");
+
+        assertUpdate(format("INSERT INTO %s VALUES %s", tableName, allValues), firstBucketKeys.size() + secondBucketKeys.size());
+
+        assertQuery(format("SELECT COUNT(*) FROM \"%s$partitions\"", tableName), "SELECT 2");
+
+        assertQuery(select + " WHERE d_bucket = 0",
+                format("VALUES(0, %d, %s, %s, %d, %d)",
+                        firstBucketKeys.size(),
+                        firstBucketMinKey,
+                        firstBucketMaxKey,
+                        firstBucketValues.stream().mapToInt(v -> v).min().getAsInt(),
+                        firstBucketValues.stream().mapToInt(v -> v).max().getAsInt()));
+
+        assertQuery(select + " WHERE d_bucket = 1",
+                format("VALUES(1, %d, %s, %s, %d, %d)",
+                        secondBucketKeys.size(),
+                        secondBucketMinKey,
+                        secondBucketMaxKey,
+                        secondBucketValues.stream().mapToInt(v -> v).min().getAsInt(),
+                        secondBucketValues.stream().mapToInt(v -> v).max().getAsInt()));
+        dropTable(tableName.toLowerCase(Locale.ENGLISH));
+    }
 
+    @Test
+    public void testApplyFilterWithNonEmptyConstraintPredicate()
```

Comment:
not sure why this method changed the name, and what the new name is expected to mean

Reply:
It didn't change name, I simply extracted part of the previous test to a specific one. 
I don't really understand why it was part of the `testBucketTransform` method. 
It was marked with this comment:
`// Exercise IcebergMetadata.applyFilter with non-empty Constraint.predicate, via non-pushdownable predicates`